### ISSUE TITLE
Adding authentication via phantom wallet then using session cookies

### ISSUE THREAD

**Issue by s1846118** [CodeDaddy69/not-a-frontend-app#8]
We want to add some form of authentication via the flask server so that a user can access more private information...
To initially authenticate we could use the phantom wallet I am sure to sign a message and authenticate the message on the server side!
Then from there we would want to look at issuing session cookies the the user after authentication so that they can see this private data for the whole time they are using the site given that we have authenticated....

- How to send cookies via flask and front end?

**Comment by CodeDaddy69** [comment]
I was thinking how do other websites - such as nft marketplaces (magiceden.io for example) do it?
There is no point where a user has to sign messages, maybe they get someones nfts directly from chain? Probably.

I think setting auth via cookies or whatnot for a session length is defo the way to go. I dont think we can query it all from chain.

**Comment by s1846118** [comment]
Yeah for sure can not query all from chain. This article describes two ways we can do it:
- https://blog.loginradius.com/engineering/cookie-based-vs-cookieless-authentication/#:~:text=The%20entire%20cookie%2Dbased%20authentication,by%20querying%20the%20user%20data. 
The second method via cookieless seems like it might fit well with a phantom wallet which has a private and public key already with a given user. This would allow us to issue these JSON tokens (see link) which basically act as a cookie given that the user has managed to authenticate themselves with their private key...

**Comment by s1846118** [comment]
Yeah good question I will look into how the other platforms do it... I just can't imagine they make people authenticate via a login/password when you can do it via the wallet!

**Comment by s1846118** [comment]
If we can add some functionality which just automatically signs some message while you are connecting your wallet and sends this to our backend then this might be a nice and simple way to authenticate without the users to have to authenticate via the blockchain and our server...

**Comment by s1846118** [comment]
Ok been looking into this a fair bit and looks like right now in production we need to use JWT. The way session cookies works with the flask server running on a different port to the front end is causing some issues (I think CORS related) which stop Flask from saving session data between requests! 

For now while in production I am going to use JWT to solve this issue (all data stored in the JWT) so no issue with Flask forgetting session data! Just need to ensure nothing private stored in the token and need to be careful as they are vulnerable to CSRF attacks.

**Comment by s1846118** [comment]
I think this is basically done. No doubt we shall return to this as we use the JWT and not sure about how safe/scalable this solution is? Will leave open for now and we can return to this once the website is at a further stage!
